src/features/locators/locators.slice.ts

```diff
@@ -49,26 +49,23 @@ const locatorsSlice = createSlice({
       locatorsAdapter.addMany(state, payload);
     },
     changeLocatorAttributes(state, { payload }: PayloadAction<ChangeLocatorAttributesPayload>) {
-      const { type, name, locator, element_id, message, isCustomName, locatorType } = payload;
+      const { locator, element_id, locatorType, ...rest } = payload;
+
       const _locator = simpleSelectLocatorById(state, element_id);
+
       if (!_locator) return;
 
-      const { fullXpath, robulaXpath } = _locator.locator;
-      const newValue = { ..._locator, locator: { ..._locator.locator }, message, type, name, isCustomName };
+      const newValue = { ..._locator, locator: { ..._locator.locator, output: locator }, locatorType, ...rest };
 
-      if (fullXpath !== locator && robulaXpath !== locator) {
-        if (locatorType !== LocatorType.cssSelector) {
-          newValue.locator.customXpath = locator;
-        } else {
-          newValue.locator.cssSelector = locator;
-        }
-        newValue.isCustomLocator = true;
-        newValue.locator.taskStatus = LocatorTaskStatus.SUCCESS;
-      } else {
-        delete newValue.locator.customXpath;
-        newValue.isCustomLocator = false;
+      if (_locator.locator.cssSelector !== locator && locatorType === LocatorType.cssSelector) {
```

Comment:
I believe LocatorType comparison would be enough. The would be easier that way

Reply:
fixed

Reply:
as far as I can see, we'll overwrite with the same value. What's the problem with that? now it's hard to understand why you don't allow to do that

Reply:
thank you!